xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xpn.xwiki.web;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.job.Job;
+import org.xwiki.job.JobException;
+import org.xwiki.job.JobExecutor;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.refactoring.internal.job.DeleteJob;
+import org.xwiki.refactoring.job.EntityRequest;
+import org.xwiki.refactoring.job.PermanentlyDeleteRequest;
+import org.xwiki.refactoring.job.RefactoringJobs;
+import org.xwiki.refactoring.script.RefactoringScriptService;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.api.DeletedDocument;
+import com.xpn.xwiki.doc.XWikiDeletedDocument;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Action for delete document to recycle bin and for delete documents from recycle bin.
+ *
+ */
+public class DeleteAction extends XWikiAction
+{
+    /**
+     * @param entityReference the entity to delete
+     * @param context the current context
+     * @return {@code true} if the user is redirected, {@code false} otherwise
+     * @throws XWikiException if anything goes wrong during the document deletion
+     * @deprecated since 12.8RC1, use {@link #deleteDocument(EntityReference, XWikiContext, boolean)} instead
+     */
+    @Deprecated
+    protected boolean deleteToRecycleBin(EntityReference entityReference, XWikiContext context)
```

Comment:
hmmm that's not how I'd have done it since and I don't understand how it can work. I was expecting to see an aspect since it's about extracting a method from an existing class.

Reply:
what you've done should break the build with revapi screaming normally.